pkg/sdkserver/localsdk.go

```diff
@@ -620,27 +620,164 @@ func (l *LocalSDKServer) UpdateCounter(ctx context.Context, in *alpha.UpdateCoun
-			return &alpha.Counter{Name: name, Count: &l.gs.Status.Counters[name].Count, Capacity: &l.gs.Status.Counters[name].Capacity}, nil
+		// Create *alpha.Counter from *sdk.GameServer_Status_CounterStatus for merging.
+		tmpCounter := &alpha.Counter{Name: name, Count: &counter.Count, Capacity: &counter.Capacity}
+		// Removes any fields from the request object that are not included in the FieldMask Paths.
+		fmutils.Filter(in.Counter, in.UpdateMask.Paths)
+		// Now that the request is filtered we can merge it with the Counter entity.
+		proto.Merge(tmpCounter, in.Counter)
+		// Verify that 0 <= Count >= Capacity
+		if *tmpCounter.Count < 0 || *tmpCounter.Count > *tmpCounter.Capacity {
+			return nil, errors.Errorf("OUT_OF_RANGE. Count must be within range [0,Capacity]. Found Count: %d, Capacity: %d", *tmpCounter.Count, *tmpCounter.Capacity)
 		}
-		return nil, errors.Errorf("INVALID_ARGUMENT. Field Mask Path(s) %v are invalid for Counter. Use valid %v", in.UpdateMask.GetPaths(), counter.ProtoReflect().Descriptor().Fields())
+		// TODO: Verify that Capacity is within bounds determined by CRD.
+		// Write newly updated Counter to gameserverstatus.
+		l.gs.Status.Counters[name].Count = *tmpCounter.Count
+		l.gs.Status.Counters[name].Capacity = *tmpCounter.Capacity
+		return &alpha.Counter{Name: name, Count: &l.gs.Status.Counters[name].Count, Capacity: &l.gs.Status.Counters[name].Capacity}, nil
 	}
 	return nil, errors.Errorf("NOT_FOUND. %s Counter not found", name)
 }
 
+// GetList returns a List. Returns NOT_FOUND if the List does not exist.
+// [Stage:Alpha]
+// [FeatureFlag:CountsAndLists]
+func (l *LocalSDKServer) GetList(ctx context.Context, in *alpha.GetListRequest) (*alpha.List, error) {
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
+	}
+
+	l.logger.WithField("name", in.Name).Info("Getting List")
+	l.recordRequest("getlist")
+	l.gsMutex.RLock()
+	defer l.gsMutex.RUnlock()
+
+	if list, ok := l.gs.Status.Lists[in.Name]; ok {
+		return &alpha.List{Name: in.Name, Capacity: &list.Capacity, Values: list.Values}, nil
+	}
+	return nil, errors.Errorf("NOT_FOUND. %s List not found", in.Name)
+}
+
+// UpdateList returns the updated List. Returns NOT_FOUND if the List does not exist (name cannot be updated).
+// **THIS WILL OVERWRITE ALL EXISTING LIST.VALUES WITH REQUEST LIST.VALUES** use AddListValue() or
+// RemoveListValue() for modifying the List.Values field.
+// Returns INVALID_ARGUMENT if the field mask path(s) are not field(s) of the List.
+// [Stage:Alpha]
+// [FeatureFlag:CountsAndLists]
+func (l *LocalSDKServer) UpdateList(ctx context.Context, in *alpha.UpdateListRequest) (*alpha.List, error) {
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
+	}
+
+	l.logger.WithField("name", in.List.Name).Info("Updating List")
+	l.recordRequest("updatelist")
+	l.gsMutex.RLock()
```

Comment:
I think we should use `Lock()` here to prevent a dirty read of `l.gs.Status.Lists`. Same for the `UpdateCounter`